src/main/java/sirpotato/Parser.java

```diff
@@ -0,0 +1,111 @@
+package sirpotato;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDate;
+
+/**
+ * Contains methods to parse user input and validate user input
+ */
+class Parser {
+
+    /**
+     * Validates the date that user inputs
+     * If not in dd-MM-yyyy format, will throw an exception
+     * @param date the date(in string format) that the user types in
+     */
+    public static void validateDateFormat(String date) throws DukeException {
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+        try {
+            LocalDate.parse(date, dateFormatter);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Mate, the date must be in the format dd-MM-yyyy.");
+        }
+    }
+
+    /**
+     * Validates user input
+     * @param userInput The user's input to the chatbot
+     */
+    public static void checkForErrors(String userInput) throws DukeException {
+        if (userInput.startsWith("todo")) {
+            if (userInput.length() <= 5 || userInput.substring(5).isEmpty()) {
+                throw new DukeException("Mate, you have got to give us a description of the task");
+            }
+        } else if (userInput.startsWith("deadline")) {
+            String[] sectionedString = userInput.split("/by ");
+            if (sectionedString.length < 2 || 
+                sectionedString[0].substring(9).isEmpty() || 
+                sectionedString[1].isEmpty()) {
+                throw new DukeException("Mate, a deadline must include a task, and the deadline");
+            }
+            validateDateFormat(sectionedString[1].trim());
+        } else if (userInput.startsWith("event")) {
+            String[] sectionedString = userInput.split(" /from | /to ");
+            if (sectionedString.length < 3 || sectionedString[0].substring(6).isEmpty() ||
+                sectionedString[1].isEmpty() || sectionedString[2].isEmpty()) {
+                throw new DukeException("Mate, an event should have the description, the start, and the end.");
+            }
+            validateDateFormat(sectionedString[1].trim());
+            validateDateFormat(sectionedString[2].trim());
+        } else if (userInput.startsWith("delete")) {
+            if (userInput.length() <= 7) {
+                throw new DukeException("You need to say which item to delete");
+            }
+        } else if (!userInput.equals("bye") && !userInput.equals("list") && 
+                   !userInput.startsWith("mark") && !userInput.startsWith("unmark")
+                   && !userInput.startsWith("find")) {
+            throw new DukeException("I'm sorry, that is not a valid input");
+        }
```

Comment:
It is good that the method's name is a noun, although for the method body it would be great if there are some whitespaces between if-else statements to improve readability

Reply:
Thanks for the suggestion - that's a good point!